src/map/script.c

```diff
@@ -9688,10 +9688,23 @@ BUILDIN(setgroupid) {
 
 /// Returns the group ID of the player.
 ///
-/// getgroupid() -> <int>
+/// getgroupid({<account ID>/<character ID>/<character name>})
 BUILDIN(getgroupid)
 {
-	struct map_session_data *sd = script->rid2sd(st);
+	struct map_session_data *sd = NULL;
+
+	/* check if a character name is specified */
+	if (script_hasdata(st, 2)) {
+		if (script_isstringtype(st, 2)) {
+			sd = map->nick2sd(script_getstr(st, 2));
+		} else {
+			int id = script_getnum(st, 2);
+			sd = (map->id2sd(id) ? map->id2sd(id) : map->charid2sd(id));
```

Comment:
it is possible that if a character has the ID of `2000000` (in the unlikely but possible future where a server has over 2000000 characters), id would be used to search the account id instead of charid?